airbyte-cdk/python/airbyte_cdk/sources/file_based/stream/default_file_based_stream.py

```diff
@@ -65,17 +72,49 @@ def read_records_from_slice(self, stream_slice: StreamSlice) -> Iterable[Mapping
         for file in stream_slice["files"]:
             # only serialize the datetime once
             file_datetime_string = file.last_modified.strftime("%Y-%m-%dT%H:%M:%SZ")
+            n_skipped = line_no = 0
+
             try:
                 for record in parser.parse_records(file, self._stream_reader):
+                    line_no += 1
+
                     if not self.record_passes_validation_policy(record):
-                        logging.warning(f"Record did not pass validation policy: {record}")
+                        n_skipped += 1
                         continue
                     record[self.ab_last_mod_col] = file_datetime_string
                     record[self.ab_file_name_col] = file.uri
                     yield stream_data_to_airbyte_message(self.name, record)
                 self._cursor.add_file(file)
+
+            except StopSyncPerValidationPolicy:
+                yield AirbyteMessage(
+                    type=Type.LOG,
+                    log=AirbyteLogMessage(
+                        level=Level.INFO,
+                        message=f"Stopping sync in accordance with the configured validation policy. Records in file did not conform to the schema. stream={self.name} file={file.uri} validation_policy={self.config.validation_policy.name}",
+                    ),
+                )
+                break
+
             except Exception as exc:
-                raise RecordParseError(FileBasedSourceError.ERROR_PARSING_FILE, stream=self.name) from exc
+                yield AirbyteMessage(
+                    type=Type.LOG,
+                    log=AirbyteLogMessage(
+                        level=Level.ERROR,
+                        message=f"{FileBasedSourceError.ERROR_PARSING_RECORD.value} stream={self.name} file={file.uri} line_no={line_no}",
+                        stack_trace="\n".join(traceback.format_exception(etype=type(exc), value=exc, tb=exc.__traceback__)),
+                    ),
+                )
+
+            else:
```

Comment:
We are only printing the `n_skipped` AirbyteMessage if there are no errors but could there be errors and skipped records? If that's the case, do we want to send both AirbyteMessages?